Source/gg2/Objects/Weapons/Revolver.events/User Event 3.xml

```diff
@@ -13,7 +13,8 @@
       <not>false</not>
       <appliesTo>.self</appliesTo>
       <arguments>
-        <argument kind="STRING">ammoCount = max(0, ammoCount-1);
+        <argument kind="STRING">if !mouse_check_button(mb_right) {
```

Comment:
Using mb_right isn't the most important issue here: The code modification is in the wrong place entirely.

Revolver User Event  3 is run on the server as well as all clients, not just on the client which controls this particular spy. You'll want to check if the *player of this spy* has his "special" input pressed, not *the client running this code*. The way to do this in the revolver code is via `owner.keyState`. It's not documented well, unfortunately (in fact I just looked up the documenting comment in `PlayerControl.BeginStep` and it's just plain wrong), but the special input is mapped to bit 3 of `keyState`, so the code should be `if(owner.keyState & $08)`.

However, that should go to `User Event 1` of Revolver, not `User Event 3`, because `User Event 3` is the synchronized "shoot" event (yeah, this is all really well documented, I know.) There is no good reason to have the server send "shoot" to all clients only for them to then decide that it won't happen after all.